pkg/contexts/ocm/transfer/transfer.go

```diff
@@ -125,25 +125,27 @@ func CopyVersion(printer common.Printer, log logging.Logger, hist common.History
 	log.Info("  transferring resources")
 	for i, r := range src.GetResources() {
 		var m ocm.AccessMethod
+
 		a, err := r.Access()
 		if err == nil {
 			m, err = r.AccessMethod()
-			if err == nil {
-				defer m.Close()
-				ok := a.IsLocal(src.GetContext())
-				if !ok {
-					if !none.IsNone(a.GetKind()) {
-						ok, err = handler.TransferResource(src, a, r)
-						if !ok {
-							log.Info("transport omitted", "resource", r.Meta().Name, "index", i, "access", a.GetType())
-						}
+		}
+		if err == nil {
+			ok := a.IsLocal(src.GetContext())
+			if !ok {
+				if !none.IsNone(a.GetKind()) {
+					ok, err = handler.TransferResource(src, a, r)
+					if err == nil && !ok {
+						log.Info("transport omitted", "resource", r.Meta().Name, "index", i, "access", a.GetType())
 					}
 				}
-				if ok {
-					hint := ocmcpi.ArtifactNameHint(a, src)
-					printArtifactInfo(printer, log, "resource", i, hint)
-					err = handler.HandleTransferResource(r, m, hint, t)
-				}
+			}
+			if ok {
+				hint := ocmcpi.ArtifactNameHint(a, src)
+				printArtifactInfo(printer, log, "resource", i, hint)
+				err = errors.Join(err, handler.HandleTransferResource(r, m, hint, t))
+			} else {
+				err = errors.Join(err, m.Close())
```

Comment:
`m` is not closed in all cases, only here. So if it is `ok` it won't be closed. Why not put this at the end? Since we only ever return at the end.

Reply:
In the ok==true case the handler is called. And here the contract says, that the given access (m) is closed.
That was the reason, why m was closed twice. 

So, for ok==true the responsibility to close the access is passed to another entity (the handler), and
for the other case it has to be closed explicitly.

I agree, that is not really nice.

Reply:
Ah, you mean this bit?
```go
func (b *blobAccess[T]) Close() error {
	b.lock.Lock()
	defer b.lock.Unlock()
	if !b.closed.Load() {
		tmp := b.access
		b.closed.Store(true)
		return tmp.Close()
	}
	return ErrClosed
}
```

I see. Well, this isn't ideal, indeed.

Reply:
I meant this implementation-induced take-over of the ownership of the access. I've changed this, now the caller (the transfor code) opens the access and has to close it again, so, the `Close` call only in the `else` part of an `if` can now be avoided. The access has always to be closed. The handler does not implicitly close it, now.